src/NLog/Targets/FileTarget.cs

```diff
@@ -1058,22 +1052,21 @@ private void RollArchivesForward(string fileName, string pattern, int archiveNum
             string newFileName = ReplaceNumberPattern(pattern, archiveNumber);
             RollArchivesForward(newFileName, pattern, archiveNumber + 1);
 
-            var shouldCompress = archiveNumber == 0;
-            try
-            {
-                RollArchiveForward(fileName, newFileName, shouldCompress);
-            }
-            catch (IOException)
```

Comment:
> why is the catch removed?

I believe the intention of the catch is to create the directory if it does not exist and then retry.

There are two cases: 1) calling `ArchiveFile()` when `archiveNumber == 0` and 2) calling `RollArchiveForward()` otherwise:  
1) `ArchiveFile()` already creates the directory if it does not exist. 
2) `RollArchiveForward()` is called for archive files only. Here `fileName` is an existing archive file, and `newFileName` is its new file path. But all archive files are in the same directory, so the new file path's directory already exists.